http/server_test.ts

```diff
@@ -1372,6 +1372,21 @@ Deno.test("serve - onListen callback is called when the server started listening
   });
 });
 
+Deno.test("serve - onListen callback is called with ephemeral port", () => {
+  const abortController = new AbortController();
+  return serve((_) => new Response("hello"), {
+    port: 0,
+    async onListen({ hostname, port }) {
+      const responseText = await (await fetch(`http://localhost:${port}/`))
```

Comment:
It might be worth having an assertion that `port` is not 0 before this `fetch()` call just to make debugging a bit easier if this ever regresses.